server/README.md

```diff
@@ -0,0 +1,181 @@
+Note: Please refer [Policy Language](https://www.openpolicyagent.org/docs/latest/policy-language/) for more about `.rego` syntax information.
+
+## API
+
+Attestation Server provides two groups of GRPC endpoints:
+- attestation: Attest the received Evidence which is defined in [protobuf](https://github.com/confidential-containers/attestation-service/server/proto/attestation.proto).
+- management: Customize the Server's configuration which is defined in [protobuf](https://github.com/confidential-containers/attestation-service/server/proto/management.proto).
+
+### Attestation
+
+The `attestation` group only includes one `Attestation` GRPC endpoint currently. It receives the attestation [Evidence](https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestation), executes attestation procedures, and responses the [Attestation Results](https://github.com/confidential-containers/attestation-service/issues/1).
+
+`Attestation` message:
+```PROTO
+message AttestationRequest {
+    // Attestation evidence: https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestatio
+    bytes evidence = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+### Management
+
+It's mainly used to customize Attestation Server's configurations:
+- Customize each type of TEE's `Policy(.rego)` and `Reference Data(.json)` in order to evaluate it's TCB status precisely.
+    - GetPolicy
+    - GetReferenceData
+    - SetPolicy
+    - SetReferenceData
+    - ResetPolicy
+    - ResetReferenceData
+- Test/Evaluate the local `Policy(.rego)` or `Reference Data(.json)` with remote Attestation Server's OPA engine.
+    - TestOpa
+
+The supported types of TEE:
+```PROTO
+enum Tee {
+    Sgx = 0;
+    Tdx = 1;
+    SevSnp = 2;
+    Sample = 3;
+}
+```
+
+#### GetPolicy
+
+Get the specific TEE's OPA `Policy(.rego)`. The message:
+```PROTO
+message GetPolicyRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+#### GetReferenceData
+
+Get the specific TEE's OPA `Reference Data(.json)`. The message:
+```PROTO
+message GetReferenceDataRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+#### SetPolicy
+
+Set the specific TEE's OPA `Policy(.rego)`. It can make the Attestation Server evaluate the TEE TCB status according to specific user's preference. And it will echo error if the new `Policy(.rego)` syntax is illegal. The message:
```

Comment:
So we're going to parse each policy file when it is uploaded to make sure the syntax is correct? Is only one policy allowed per tee? Does uploading a new one overwrite an old one?

Reply:
> So we're going to parse each policy file when it is uploaded to make sure the syntax is correct? 

Yes, I think the policy file's syntax should be checked, or it will make the evaluation unworkable.

> Is only one policy allowed per tee? Does uploading a new one overwrite an old one?

Yes, the current implementation is one policy per TEE. 
Yes, the upload new one will cover the old one directly and the old one won't be stored by AS (except the default policy.rego because it is intergated inside AS crate.)

Reply:
I guess that's fine. It might be nice to be able to store multiple policies even if only one is active. That way you could easily switch between policies without having to upload them again. We can add that later, though.